src/core/thread/mle.cpp

```diff
@@ -1098,6 +1095,8 @@ void Mle::SetCslTimeout(uint32_t aTimeout)
 exit:
     return;
 }
+
+bool Mle::IsCslSupported(void) const { return IsChild() && GetParent().IsEnhancedKeepAliveSupported(); }
```

Comment:
Should we do `IsThreadVersion1p2OrHigher()` directly instead of `IsEnhancedKeepAliveSupported()` (I feel the check for enh keep alive is unrelated and an indirect way of checking version?)